src/Lecture1.hs

```diff
@@ -29,98 +29,38 @@ module Lecture1
     , subString
     , strSum
     , lowerAndGreater
-    ) where
+	, check
+    ) where 
 
--- VVV If you need to import libraries, do it after this line ... VVV
-
--- ^^^ and before this line. Otherwise the test suite might fail  ^^^
-
-{- | Specify the type signature of the following function. Think about
-its behaviour, possible types for the function arguments and write the
-type signature explicitly.
--}
+makeSnippet::Int -> String -> String
 makeSnippet limit text = take limit ("Description: " ++ text) ++ "..."
 
-{- | Implement a function that takes two numbers and finds sum of
-their squares.
-
->>> sumOfSquares 3 4
-25
-
->>> sumOfSquares (-2) 7
-53
-
-Explanation: @sumOfSquares 3 4@ should be equal to @9 + 16@ and this
-is 25.
--}
--- DON'T FORGET TO SPECIFY THE TYPE IN HERE
-sumOfSquares x y = error "TODO!"
+sumOfSquares:: Int -> Int -> Int
+sumOfSquares x y = (x^2) + (y ^2)
 
-{- | Implement a function that returns the last digit of a given number.
+lastDigit::Int->Int
+lastDigit x = abs $ x `rem` 10
 
->>> lastDigit 42
-2
->>> lastDigit (-17)
-7
+minmax::Int -> Int -> Int-> Int
+minmax x y z = let (max,min) = (maximum [x,y,z],minimum [x,y,z]) in (max - min)
```

Comment:
I see that GHC outputs some warnings ⚠️
In this case, it happens because `max` and `min` are the names of standard functions in Prelude. It still works due to name shadowing. But generally, it's better to avoid name conflicts with standard functions.

It's a bit tough to do this here because both `max` and `maximum` are already taken 😅 But I would use smth like `mx` or `maxVal` (and similar for minimum).